Linaro/openssl-uadk
# Add rsa engine function

Cipher Test:
openssl engine -c uadk
openssl enc -aes-128-cbc -engine uadk -pass pass:123456
openssl enc -aes-192-cbc -engine uadk -pass pass:123456
openssl enc -aes-256-cbc -engine uadk -pass pass:123456
openssl enc -aes-128-ctr -engine uadk -pass pass:123456
openssl enc -aes-192-ctr -engine uadk -pass pass:123456
openssl enc -aes-256-ctr -engine uadk -pass pass:123456

md5 test:
openssl speed -engine uadk hmac

Fix:
1. uadk will install head files in /usr/local/include/uadk, so let use correct
head file, otherwise data will be wrong.
2. replace ctx to sess.
3. rename uadk_ras_sess.
4. fix ctx_mode bug.
5. introduce error print function.
6. add callback functions for genkey, enc, dec, sign, verify.
7. delete uadk_rsa_crypto(), use uadk_rsa_sync() instead.
8. clean uadk_rsa_fill_prikey(),add uadk_rsa_fill_prikey_crt().
9. rsa: remove init/destroy to e_uadk.c.
10. fix resource init & uninit.
11. remove no need macro.
12. make functions to static.
13. fix ENGINE_set_RSA() input.
14. add some missing functions.
15. add pubkey check function.

RSA functions:
So far, RSA-keygen,RSA-encrypt,RSA-decrypt,RSA-sign,RSA-verify SYNC MODE functions are OK.
The test commands:
1. Generate RSA private key
openssl genrsa -out prikey.pem -engine uadk 2048
2. Genetrate RSA public key
openssl rsa -in prikey.pem -pubout -out pubkey.pem -engine uadk
3. RSA encrypt
openssl rsautl -encrypt -in plain.txt -inkey pubkey.pem -pubin -out enc.txt -engine uadk
4. RSA decrypt
openssl rsautl -decrypt -in enc.txt -inkey prikey.pem -out dec.txt -engine uadk
5. RSA sign
openssl rsautl -sign -in msg.txt -inkey prikey.pem -out signed.txt -engine uadk
6. RSA verify
openssl rsautl -verify -in signed.txt -inkey pubkey.pem -pubin -out verified.txt -engine uadk
7. Speed
openssl speed -elapsed -engine uadk rsa2048

-RSA engine--- Edit by: Zhou Wang, Zhiqi Song.

Signed-off-by: Zhiqi Song <songzhiqi1@huawei.com>

fix coding style

Signed-off-by: Zhiqi Song <songzhiqi1@huawei.com>
